featurebyte/routes/historical_feature_table/controller.py

```diff
@@ -49,108 +54,49 @@ def __init__(
-            request_column_names = set(observation_set_dataframe.columns)
+        table_create: TableCreateT,
+        observation_set_dataframe: Optional[pd.DataFrame],
+    ) -> PayloadT:
+        assert isinstance(table_create, HistoricalFeatureTableCreate)
+        return await self.service.get_historical_feature_table_task_payload(  # type: ignore[return-value]
+            data=table_create, observation_set_dataframe=observation_set_dataframe
+        )
 
+    async def get_validation_parameters(self, table_create: TableCreateT) -> ValidationParameters:
+        assert isinstance(table_create, HistoricalFeatureTableCreate)
         # feature cluster group feature graph by feature store ID, only single feature store is
         # supported
-        feature_cluster = data.featurelist_get_historical_features.feature_clusters[0]
+        feature_cluster = table_create.featurelist_get_historical_features.feature_clusters[0]
         feature_store = await self.feature_store_service.get_document(
             document_id=feature_cluster.feature_store_id
         )
-        await self.entity_validation_service.validate_entities_or_prepare_for_parent_serving(
+        return ValidationParameters(
             graph=feature_cluster.graph,
             nodes=feature_cluster.nodes,
-            request_column_names=request_column_names,
             feature_store=feature_store,
-            serving_names_mapping=data.featurelist_get_historical_features.serving_names_mapping,
-        )
-
-        # prepare task payload and submit task
-        payload = await self.service.get_historical_feature_table_task_payload(
-            data=data, observation_set_dataframe=observation_set_dataframe
+            serving_names_mapping=table_create.featurelist_get_historical_features.serving_names_mapping,
         )
-        task_id = await self.task_controller.task_manager.submit(payload=payload)
-        return await self.task_controller.get_task(task_id=str(task_id))
 
-    async def get_info(self, document_id: ObjectId, verbose: bool) -> HistoricalFeatureTableInfo:
-        """
-        Get HistoricalFeatureTable info
-
-        Parameters
-        ----------
-        document_id: ObjectId
-            HistoricalFeatureTable ID
-        verbose: bool
-            Whether to return verbose info
-
-        Returns
-        -------
-        HistoricalFeatureTableInfo
-        """
-        _ = verbose
-        historical_feature_table = await self.service.get_document(document_id=document_id)
-        if historical_feature_table.observation_table_id is not None:
-            observation_table = await self.observation_table_service.get_document(
-                document_id=historical_feature_table.observation_table_id
-            )
-        else:
-            observation_table = None
+    async def get_additional_info_params(
+        self, document: MaterializedTableDocumentT
```

Comment:
Can `MaterializedTableDocumentT` be replaced `HistoricalFeatureTableModel` in the type annotation here? The type parameter `MaterializedTableDocumentT` is already tied to `HistoricalFeatureTableModel` based on the class definition:

```
class HistoricalFeatureTableController(
    FeatureOrTargetTableController[
        HistoricalFeatureTableModel,
        HistoricalFeatureTableService,
        HistoricalFeatureTableList,
        HistoricalFeatureTableInfo,
    ],
):
   ...
```

Reply:
hm yea, let me try again. i was getting some lint errors on a previous iteration.